### PR TITLE
Angular 13 support

### PR DESCRIPTION
**Pull request by david-shortman**
- Migrates to Angular 13 support
- Resolves https://github.com/microsoft/powerbi-client-angular/issues/15

Following this PR, I will open an Angular 14 version packaged at v4.0.0

### CODE REVIEW

**Comment by david-shortman** [comment]
@ali-hamud @may-hartov I'm seeking a maintainer to review this Angular upgrade. Thank you.

File: Angular/package.json [hunk 1]
```diff
@@ -42,14 +42,14 @@
...
     "eslint-plugin-prefer-arrow": "1.2.2",
-    "jasmine-core": "~3.6.0",
+    "jasmine-core": "~3.8.0",
     "jasmine-spec-reporter": "~5.0.0",
-    "karma": "~6.1.0",
+    "karma": "~6.4.0",
     "karma-chrome-launcher": "~3.1.0",
     "karma-coverage": "~2.0.3",
     "karma-jasmine": "~4.0.0",
     "karma-jasmine-html-reporter": "^1.5.0",
```
Comment (KotanaSai21) [inline Angular/package.json:51]: "karma-jasmine-html-reporter": "^1.7.0",
Can't we also karma-jasmine-html-report to the latest stable version ?
image

**Comment by david-shortman** [comment]
@orshemesh16 I am seeking review for this update

File: Angular/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "angular",
-  "version": "3.1.0",
+  "version": "13.0.0",
```
[the remark this answers is not in the document]
Comment (orshemesh16) [reply Angular/package.json:3]: In addition, we should update the package version as well to `3.0.0` (update the major):
[powerbi-client-angular](https://github.com/microsoft/powerbi-client-angular)/[Angular](https://github.com/microsoft/powerbi-client-angular/tree/main/Angular)/[powerbi-client-angular](https://github.com/microsoft/powerbi-client-angular/tree/main/Angular/powerbi-client-angular)/package.json

**Review by orshemesh16 — CHANGES_REQUESTED** [review]

File: Angular/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (david-shortman) [reply Angular/package.json]: Updated both. Thanks!

**Review by orshemesh16 — APPROVED** [review]

File: Angular/powerbi-client-angular/config/src/tsconfig.lib.prod.json [hunk 3]
```diff
@@ -4,6 +4,6 @@
     "declarationMap": false
   },
   "angularCompilerOptions": {
-    "compilationMode": "partial"
+    "compilationMode": "full"
```
[the remark this answers is not in the document]
Comment (david-shortman) [reply Angular/powerbi-client-angular/config/src/tsconfig.lib.prod.json:7]: This fulfills https://github.com/microsoft/powerbi-client-angular/issues/15, which drops support for the ViewEngine and exclusively compiles for Ivy.

File: Angular/powerbi-client-angular/ng-package.json [hunk 4]
```diff
@@ -2,11 +2,7 @@
   "$schema": "../node_modules/ng-packagr/ng-package.schema.json",
   "dest": "../dist/powerbi-client-angular",
   "lib": {
-    "entryFile": "src/public-api.ts",
-    "umdModuleIds": {
```
[the remark this answers is not in the document]
Comment (david-shortman) [reply Angular/powerbi-client-angular/ng-package.json:6]: This was automatically migrated by the Angular upgrade CLI tool

File: Angular/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (david-shortman) [reply Angular/package.json]: This is the version of the demo app. I'm not sure how to semantically version it, so I left it at `3.1.0`.

**Pull request merged**